visualization/app/codeCharta/services/isAttributeSideBarVisible.service.ts

```diff
@@ -0,0 +1,16 @@
+import { ThreeSceneService } from "../ui/codeMap/threeViewer/threeSceneService"
+import { Inject, Injectable } from "@angular/core"
+
+@Injectable({ providedIn: "root" })
+export class IsAttributeSideBarVisibleService {
+	isOpen = false
+
+	constructor(@Inject(ThreeSceneService) private threeSceneService: ThreeSceneService) {
+		this.threeSceneService.subscribe("onBuildingSelected", () => {
+			this.isOpen = true
+		})
+		this.threeSceneService.subscribe("onBuildingDeselected", () => {
+			this.isOpen = false
+		})
+	}
+}
```

Comment:
Why did you choose that 'attritbute sidebar is open' property should be in a service? I am just curious.

Reply:
good question! Previous we had isAttributeSideBarVisbile in the store, which was always set, when appStatus.selectedBuildingId was set in the store. But the store should only contain atomic values and isAttributeSideBarVisbile is obvious derived from selectedBuildingId. Naming can probably be further improved, but better as before... :)